tests/console/yast2_samba.pm

```diff
@@ -14,258 +14,338 @@ use strict;
+    wins_support           => 'Yes',
+    inherit_acls           => 'Yes',
+    read_only              => 'Yes'
+);
+
+sub smb_conf_checker {
+    my $error = "";
+    # select global & add share sections
+    my $select_script = q@
+        CONFIG='/etc/samba/smb.conf' 
+        for i in global html_public; do 
+            sed -n '/\['$i'\]/,/\[/{/^\[.*$/!p}' $CONFIG | while read -r line; do
+            printf "%-23s = %s\n" "${line%?=*}" "${line#*=?}" >> /tmp/smb.txt
+            done
+        done@;
+
+    die 'Updated smb.conf section is missing' if script_run($select_script);
+    foreach (sort keys %samba_directives) {
+        (my $new_key = $_) =~ s/_/ /g;
+        if (script_run("grep \"^$new_key *.* = $samba_directives{$_}\$\" /tmp/smb.txt")) {
+            $error .= "smb directive \"$new_key = $samba_directives{$_}\" not found in \/etc\/samba\/smb\.conf\n";
+        }
     }
 
-    zypper_call("in yast2-auth-server yast2-ldap openldap2 openldap2-client krb5-server krb5-client tdb-tools", timeout => 240);
-
-    script_run("yast2 auth-server; echo yast2-auth-server-status-\$? > /dev/$serialdev ", 0);
-    # check ldap server configuration started
-    assert_screen([qw(yast2_ldap_configuration_startup yast2_still_susefirewall2)], 60);
-    if (match_has_tag 'yast2_still_susefirewall2') {
-        record_soft_failure 'bsc#1064405';
-        send_key 'alt-c';
+    if ($error ne "") {
+        assert_script_run("echo \"$error\" > /tmp/failed_smb_directives.log");
+        die 'Missing smb.conf directives';
     }
-    # only older version like SLES 12, Leap 42.3 as well as TW should still check the needle
-    if (is_sle('<15') || is_leap('<15.0')) {
-        assert_screen 'yast2_ldap_configuration_general-setting_firewall', 60;
+}
+
+sub setup_yast2_ldap_server {
+    my %ldap_options_to_dirs = (
+        f => 'fqdn',
+        i => 'dir_instance',
+        t => 'dir_suffix',
+        d => 'dir_manager_dn',
+        r => 'dir_manager_passwd',
+        e => 'dir_manager_passwd',
+        s => 'ca_cert_pem',
+        v => 'srv_cert_key_pkcs12'
+    );
+
+    # setup FQDN hostname
+    assert_script_run("hostname openqa.ldap.test");
+    assert_script_run("echo '127.0.0.1 openqa.ldap.test openqa.ldap' > /etc/hosts");
+    # create CA cert & pkcs12 cert for LDAP
+    assert_script_run("openssl req -newkey rsa:2048 -nodes -keyout /etc/key.pem -x509 -days 365 \ "
+          . "-out /etc/certificate.pem -subj '/C=DE/ST=Bayern/L=Nuremberg/O=Suse/OU=QA/CN=localhost/emailAddress=admin\@localhost'");
+    assert_script_run("openssl pkcs12 -inkey /etc/key.pem -in /etc/certificate.pem -export -out /etc/certificate.p12 -passout pass:");
+
+    script_run("yast2 ldap-server; echo yast2-ldap-server-status-\$? > /dev/$serialdev ", 0);
+
+    # setup LDAP
+    wait_still_screen(2);
+    foreach (sort keys %ldap_options_to_dirs) {
+        wait_screen_change { send_key "alt-$_" };
+        type_string($ldap_directives{$ldap_options_to_dirs{$_}} . "\n");
```

Comment:
What's about asserting what we have typed? It's again thing we learned by running tests. For example, here we start typing, not all keys reached SUT and we already press alt-X and switch to another field.

Reply:
It is only 8 fields in one screen to be filled. Previously I have used `type_string_slow` but that should not be used in such a case. Frankly I feel like there is way too much needling in ncurses tests in general, so I have tried to use different approach. At least in my workers I have not seen any failure at this part. However, I have not paid so much attention to the LDAP part, because in SP4 it works well and distros using 389-ds are failing to restart this service due to product bug in 389.

Reply:
Yes, too many steps in the test makes it complicated. Let's then run it 100 times locally and see if this is an issue before putting extra efforts. But what's about introducing needle which verifies that we typed correctly in all fields, so in case we fail it's visible why? WDYT?